### ISSUE TITLE
Use a third-party Python 2/3 compatibility library to replace uses of twisted.python.compat.NativeStringIO

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7476]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#7476|
|Type|enhancement|
|Created|2014-06-04 17:40:13Z|
|Branch|https://github.com/twisted/twisted/tree/six-stringio-7476|

There is one in six.  We'll have to see if it actually works the way we want it, but give it a burl.

<details><summary>Searchable metadata</summary>

```
trac-id__7476 7476
type__enhancement enhancement
reporter__radix radix
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_six_stringio_7476 branches/six-stringio-7476
branch_author__radix radix
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1401903613000000 1401903613000000
changetime__1594450797273670 1594450797273670
version__None None
owner__radix radix

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

(In [42761]) Branching to six-stringio-7476.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set status to `closed`|
|-|-|

This was addressed in [#9793](https://github.com/twisted/twisted/issues/9793) by using `io.StringIO` for this purpose.
